### PR TITLE
Output external clusterings with visualisations

### PR DESCRIPTION
**Pull request by nickjcroucher**
Some fixes to the code, and added a new test (`test_beebop.py`), although not sure if this is being run with the CI? Code not perfect but better.

### CODE REVIEW

**Comment by nickjcroucher** [comment]
@EmmaLRussell has confirmed this enables beebop to produce networks annotated with external clustering.

File: PopPUNK/visualise.py [hunk 1]
```diff
@@ -419,14 +418,15 @@ def generate_visualisations(query_db,
                 mode = "lineages"
                 suffix = "_lineages.csv"
             cluster_file = os.path.join(model_prefix, os.path.basename(model_prefix) + suffix)
-        isolateClustering = readIsolateTypeFromCsv(cluster_file,
-                                                   mode = mode,
-                                                   return_dict = True)
+
+    isolateClustering = readIsolateTypeFromCsv(cluster_file,
+                                               mode = mode,
```
Comment (johnlees) [inline PopPUNK/visualise.py:423]: Given that this was an issue before, perhaps somewhere above/clearly setting a default value for mode would be sensible

File: PopPUNK/visualise.py [hunk 2]
```diff
@@ -396,13 +396,12 @@ def generate_visualisations(query_db,
...
     if external_clustering:
+        mode = 'external'
         cluster_file = external_clustering
-        isolateClustering = readIsolateTypeFromCsv(cluster_file,
-                                                   mode = 'external',
-                                                   return_dict = True)
-
+        suffix = "_clusters.csv"
+        if cluster_file.endswith('_lineages.csv'):
+            suffix = "_lineages.csv"
```
Comment (johnlees) [inline PopPUNK/visualise.py:403]: An if/else is clearer:
```suggestion
        if cluster_file.endswith('_lineages.csv'):
            suffix = "_lineages.csv"
        else:
            suffix = "_clusters.csv"
```

File: test/test-beebop.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python
+# Copyright 2018-2024 John Lees and Nick Croucher
+
+"""Tests for PopPUNK --update-db order"""
+
+import subprocess
+import os, sys
+import sys
+import shutil
```
Comment (johnlees) [inline test/test-beebop.py:9]: To get this to run in the CI you can add [here](https://github.com/bacpop/PopPUNK/blob/master/.github/workflows/azure_ci.yml#L39) or add `subprocess.run('test-beebop.py')` to `test/run_test.py`

But, I would actually suggest just copying lines 16-19 below into the end of `test/run_test.py`

**Review by johnlees — CHANGES_REQUESTED** [review]
We will also need to bump the version again

File: test/test-beebop.py (hunk 3 above)
Comment (nickjcroucher) [reply test/test-beebop.py]: Great, done in d2421f1.

File: PopPUNK/visualise.py (hunk 1 above)
Comment (nickjcroucher) [reply PopPUNK/visualise.py:424]: Agree, but can we defer to a future time?

**Comment by nickjcroucher** [comment]
Version bumped in d5179fd, thank you for the quick review!

File: PopPUNK/visualise.py (hunk 1 above)
Comment (johnlees) [reply PopPUNK/visualise.py:424]: Looking through the context to this more and I think it's fine now

**Review by johnlees — APPROVED** [review]

**Pull request merged**